include/xsimd/arch/xsimd_avx2.hpp

```diff
@@ -116,6 +117,160 @@ namespace xsimd
             }
         }
 
+        // load_masked
+        template <class A, bool... Values, class Mode>
+        XSIMD_INLINE batch<int32_t, A> load_masked(int32_t const* mem, batch_bool_constant<int32_t, A, Values...> mask, convert<int32_t>, Mode, requires_arch<avx2>) noexcept
+        {
+            XSIMD_IF_CONSTEXPR(mask.none())
+            {
+                return _mm256_setzero_si256();
+            }
+            else XSIMD_IF_CONSTEXPR(mask.all())
+            {
+                return load<A>(mem, Mode {});
+            }
+            // confined to lower 128-bit half (4 lanes) → forward to SSE
+            else XSIMD_IF_CONSTEXPR(mask.countl_zero() >= 4)
```

Comment:
Thinking of it, we could remove the need for `countl_zero` by just masking, something like

```
if((mask.mask() & 0xF0) == 0)
```

which would be a decent implementation for
```
if(mask.lower_half() == 0) // or lower_mask ?
```
which has the nice property of not encoding any "magical value"